src/tetra_rp/core/discovery.py

```
@@ -0,0 +1,409 @@
+        # Check keyword argument: resource_config=var_name
+        for keyword in decorator.keywords:
+            if keyword.arg == "resource_config":
+                if isinstance(keyword.value, ast.Name):
+                    return keyword.value.id
+
+        # Check positional argument: @remote(var_name)
+        if decorator.args and isinstance(decorator.args[0], ast.Name):
+            return decorator.args[0].id
+
+        return ""
+
+    def _import_module(self, file_path: Path):
+        """Import a Python module from file path.
+
+        Args:
+            file_path: Path to Python file
+
+        Returns:
+            Imported module or None if import fails
+        """
+        try:
+            # Create module spec
+            module_name = file_path.stem
+            spec = importlib.util.spec_from_file_location(module_name, file_path)
+
+            if not spec or not spec.loader:
+                return None
+
+            # Load module
+            module = importlib.util.module_from_spec(spec)
+            sys.modules[module_name] = module
+            spec.loader.exec_module(module)
+
+            self._scanned_modules.add(module_name)
+
+            return module
+
+        except Exception as e:
+            log.warning(f"Failed to import {file_path}: {e}")
+            return None
+
+    def _resolve_resource_variable(self, module, var_name: str) -> DeployableResource:
+        """Resolve variable name to DeployableResource instance.
+
+        Args:
+            module: Imported module
+            var_name: Variable name to resolve
+
+        Returns:
+            DeployableResource instance or None
+        """
+        try:
+            obj = getattr(module, var_name, None)
+
+            if obj and isinstance(obj, DeployableResource):
+                return obj
+
+            log.debug(
+                f"Variable '{var_name}' is not a DeployableResource (type: {type(obj)})"
+            )
+
+        except Exception as e:
```

I'm trying to think of a scenario here where we don't want to explicitly throw an err when decorating a non-deployable resource, or we fail to resolve to a deployable resource. Would it be better to just let this bubble up? Or, have a more explicit `log.warning` like `resource {var_name} failed to resolve to deployable resource, check that this is actually a deployable resource type.`